spectator-api/src/test/java/com/netflix/spectator/api/histogram/PercentileDistributionSummaryTest.java

```diff
@@ -25,19 +27,98 @@
 @RunWith(JUnit4.class)
 public class PercentileDistributionSummaryTest {
 
-  @Test
-  public void percentile() {
-    Registry r = new DefaultRegistry();
-    PercentileDistributionSummary t = PercentileDistributionSummary.get(r, r.createId("test"));
+  private Registry newRegistry() {
+    return new DefaultRegistry(Clock.SYSTEM, k -> null);
+  }
+
+  private void checkPercentiles(PercentileDistributionSummary t, int start) {
     for (int i = 0; i < 100_000; ++i) {
       t.record(i);
     }
-
-    for (int i = 0; i <= 100; ++i) {
-      double expected = i * 1e3;
+    for (int i = start; i <= 100; ++i) {
+      double expected = i * 1000.0;
       double threshold = 0.15 * expected;
       Assert.assertEquals(expected, t.percentile(i), threshold);
     }
   }
 
+  @Test
+  public void percentile() {
+    Registry r = newRegistry();
+    PercentileDistributionSummary t = PercentileDistributionSummary.get(r, r.createId("test"));
+    checkPercentiles(t, 0);
+  }
+
+  @Test
+  public void builder() {
+    Registry r = newRegistry();
+    PercentileDistributionSummary t = PercentileDistributionSummary.builder(r)
+        .withName("test")
+        .build();
+    checkPercentiles(t, 0);
+  }
+
+  @Test
+  public void builderWithThreshold() {
+    Registry r = newRegistry();
+    PercentileDistributionSummary t = PercentileDistributionSummary.builder(r)
+        .withName("test")
+        .withThreshold(100_000)
+        .build();
+    checkPercentiles(t, 25);
```

Comment:
Is this needed because the default accuracy will make it unreliable for <25?

Reply:
Yes, those will have a higher error bound because they are less than the min value that gets selected.